internal/coord/query_test.go

```diff
@@ -342,16 +340,13 @@ func TestPooledQuery_deadlock_regression(t *testing.T) {
 	// Advance the query pool state machine. Because we returned a new node
 	// above, the query pool state machine wants to send another outbound query
 	ev, _ = c.queryBehaviour.Perform(ctx)
-	require.IsType(t, &EventAddNode{}, ev) // event to notify the routing table
-	ev, _ = c.queryBehaviour.Perform(ctx)
 	require.IsType(t, &EventOutboundSendMessage{}, ev)
+	ev, _ = c.queryBehaviour.Perform(ctx)
```

Comment:
Best to use DrainBehaviour from https://github.com/plprobelab/zikade/pull/64 here. We just want to run the behaviour until it has completed all its work. Future changes might introduce new events that would affect this test.